### DISCUSSION TITLE
Engines in an Offline Environment

### DISCUSSION THREAD

**Discussion by lixaotec** [prisma/prisma#18608] [Q&A]
Please improve the instalation for an offline env.

Its very confusing trying to grab engines by have and placing them at node_modules

Even following .env custom vars, it wont work.

It seems that it is expection adding '-windows' at the middle of the file names.

thanks

**Comment by janpio** [comment]
> Its very confusing trying to grab engines by have and placing them at node_modules

Can you describe what is causing problems to you?

> Even following .env custom vars, it wont work.

Can you be more precise what does not work?

> It seems that it is expection adding '-windows' at the middle of the file names.

What do you mean by that?

**Reply by lixaotec** [reply]
Dear @jan 

Appreciate your quicly response

> Can you describe what is causing problems to you?

Well, following the instructions, whenever I need to use an 'offline' env, it falls to the option of 'custom' engines.
In fact I do not wish to use custom engines, so I just download them from prisma prebuilt, looking for my env, which is windows.
https://www.prisma.io/docs/concepts/components/prisma-engines

But when I do unzip them in a folder, and tells the .env with the paths to the binaries, running prisma always end up trying to download (by itself) those files (query_engine.dll.node.sha256 for instance).

So I did try all combinations, droping them the respective node_modules folders, and still nothing, (.prisma, prisma, @prisma) 

The only way I could achieve its running was to install the package in a working internet machine, and then copied the node_modules folders to the offline env. 

but repeat this process is kind annoying.

It would be really nice, if prisma packaging consider that some env will not have access to internet (although we have access to npm registry).

Thanks in advance

**Reply by jan** [reply]
@lixaotec wrong @jan

**Reply by lixaotec** [reply]
sorry , @janpio

**Reply by janpio** [reply]
I think that is a problem related to https://github.com/prisma/prisma/issues/12230

**Reply by lixaotec** [reply]
thanks.. 
yes.. the same question
ill end this discussion then.
